### PR TITLE
Feature/blog

### PR DESCRIPTION
**Pull request by Mateaus**
Added content:
- Post list UI
- Post content UI
- Post models
- Next and Previous content button
- Blog url routing connected with main project.
- CSS update with custom variables for font-size and font color.
Bug fixed:
- Navbar currentview couldn't find the current active link after opening a post. Now it only looks for one if there is one available.

### CODE REVIEW

File: personalwebsite/static/blog/dist/js/blog.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return string;
+    }
+
+    /* TODO: Find the closest empty char to truncate. */
+    return string.substring(0, charIndex) + ' ...';
+}
+
+function textBlock() {
+    var truncateAtIndex = 200;
+    var text = document.getElementsByClassName("text-block");
```
Comment (Anthony9698) [inline personalwebsite/static/blog/dist/js/blog.js]: Where exactly is this being used? I searched all files for the "text-block" class and couldn't find anything.

File: personalwebsite/templates/base_page_view.html [hunk 2]
```diff
@@ -35,7 +36,7 @@
             </button>
 
             <a id="my-current-view" class="navbar-brand font-medium"
-                style="margin: 0 0 0 0; padding-left: 16px; display: none;" href="">Navbar</a>
+                style="margin: 0 0 0 0; padding-left: 16px; display: none;" href=""></a>
```
Comment (Anthony9698) [inline personalwebsite/templates/base_page_view.html]: This style should probably be in its own class.

File: personalwebsite/blog/templates/blog/post_detail.html [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            {% endif %}
+
+            {% endwith %}
+            {% endwith %}
+        </div>
+    </div>
+
+    <hr>
+
+    <div class="post-content">{{ post_detail.content|safe }}</div>
```
Comment (Anthony9698) [inline personalwebsite/blog/templates/blog/post_detail.html]: Word wrapping should be enabled here for long post detail content. I think you can add this to your .post-content class.

File: personalwebsite/blog/models.py [hunk 4]
```diff
@@ -1,3 +1,33 @@
...
+
+    def __str__(self):
+        return self.name
+
+
+class Post(models.Model):
+    title = models.CharField(max_length=256)
+    overview = models.TextField()
+    tags = models.ManyToManyField(Tag)
+    estimated_time = models.IntegerField()
```
Comment (Anthony9698) [inline personalwebsite/blog/models.py:18]: I think in a future task we can automate this by using some sort of API that estimates reading time-based on word count or make our own. For now, this is fine though,

**Review by Anthony9698 — CHANGES_REQUESTED** [review]
This is a really good addition to the site and I really like the way everything is set up so far. I only have a few minor changes that I would make before merging with main.

File: personalwebsite/blog/models.py (hunk 4 above)
Comment (Mateaus) [reply personalwebsite/blog/models.py:18]: That is a good idea, I will add it to the list.

**Comment by Mateaus** [comment]
Changes:
- Fixed bug where post overview wasn't wrapping if the text was too long.
- Improved navbar, removed unnecessary js and instead now is mostly being handled by css.

**Pull request merged**